nixos/modules/services/networking/spacecookie.nix

```diff
@@ -4,9 +4,14 @@ with lib;
 
 let
   cfg = config.services.spacecookie;
-  configFile = pkgs.writeText "spacecookie.json" (lib.generators.toJSON {} {
+
+  spacecookieConfig = {
     inherit (cfg) hostname port root;
-  });
+  } // cfg.extraConfig;
+
+  configFile = pkgs.writeText "spacecookie.json"
```

Comment:
How about
```
format = pkgs.formats.json {};
configFile = formats.generate "spacecookie.json" cfg.settings;
```
and replace `extraConfig` with an option called `settings`, `type = formats.type;`.

Reply:
The approach described in <https://github.com/NixOS/rfcs/pull/42> is actually a good fit for this module and the changes in the next release of `spacecookie`.

Is it possible to add optional options without defaults, so we'd still have documentation for options under `settings`?

Reply:
Take a look in the NixOS manual for free form module options. I think you'll like what you see.